ext/coverage/coverage.c

```diff
@@ -75,6 +75,33 @@ rb_coverage_result(VALUE klass)
     return ncoverages;
 }
 
+static int
+coverage_merge_result_i(VALUE path, VALUE coverage, VALUE coverages)
+{
+    VALUE ary = rb_hash_lookup2(coverages, path, Qnil);
+
+    if (RTEST(ary)) {
+	rb_ary_replace(ary, coverage);
+    }
+
+    return ST_CONTINUE;
+}
+
+static VALUE
+rb_coverage_set_result(VALUE klass, VALUE ncoverages)
+{
+    Check_Type(ncoverages, T_HASH);
+
+    VALUE coverages = rb_get_coverages();
```

Comment:
Intermingled declaration is C99, you can't use it in the core and bundled extensions.